functions/misc/server.py

```diff
@@ -67,6 +67,17 @@ def get_memory_usage():
         print("error getting memory usage")
         os._exit(2)
 
+def get_ip_address_from_env():
+    try:
+        machine_ip = "unset"
+        if "HOST_MACHINE_IP" in os.environ.keys():
+            machine_ip = os.environ["HOST_MACHINE_IP"]
+        return machine_ip
+
+    except Exception as e:
+        print(e, file=sys.stderr)
+        print("HOST_MACHINE_IP not set")
```

Comment:
`HOST_MACHINE_IP` not being set can't be a problem as if it's unset it will get the default value so this is a confusing error message that will lead people to the wrong path debugging if it ever pops up.